src/iter/par_bridge.rs

```diff
@@ -152,65 +144,22 @@ where
         F: Folder<Self::Item>,
     {
         loop {
-            match self.items.steal() {
-                Steal::Success(it) => {
+            if let Ok(mut iter) = self.iter.lock() {
+                if let Some(it) = iter.next() {
```

Comment:
We either need to check `done` before calling this, or else `.fuse()` the iterator, so we never call the underlying iterator again after `None`.

Reply:
I wonder if we should just get rid of `done` entirely in favor of just fusing the iterator, since we're now expecting each consumer to lock for every item already. It would be simpler not to micro-optimize that end state.

Reply:
@cuviper Getting rid of `done` would require the Iterator to also be `Peekable`, so that we know when to stop `split`ting. I think that's going a little bit out of the realm of "micro-optimization"

Reply:
The splits are still limited by `split_count`, which starts as the number of threads in the pool. I think that's probably not enough to worry about optimizing away, but it's not a big deal either way.

Reply:
I agree, implemented now without `done`